### PR TITLE
Enable OTA updates for firmware through App

### LINKED ISSUE #505: Finish setting up OTA updates for firmware. ($200)
**Is your feature request related to a problem? Please describe.**
From the app I should be able to update to new firmware versions.

**Describe the solution you'd like**
There should be a device settings in the app, that determines current firmware vs somewhere hosted maybe github, and tells if there's a new firmware version available.

Then should download that zip file, and use dfu.dart (which already works) to do the firmware restart on the device for the user.

**Additional context**
`dfu.dart` was commented, because the dependency was causing building issues for iOS and android (this should be fixed too)

### PR DESCRIPTION
**Pull request by mdmohsin7**
Closes #505

### CODE REVIEW

File: backend/routers/update.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import os
```
Comment (josancamon19) [inline backend/routers/update.py]: rename to "firmware.py

File: app/lib/pages/home/firmware_update.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
+import 'package:flutter/material.dart';
```
Comment (josancamon19) [inline app/lib/pages/home/firmware_update.dart:1]: Did you test this with the compiled android and ios versions? I've found issues before with the nrf library

File: app/lib/pages/home/firmware_mixin.dart [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import 'dart:convert';
```
Comment (josancamon19) [inline app/lib/pages/home/firmware_mixin.dart:1]: How bad is it if during update, people turn off their devices, or kill the app in the middle of it, can it recover?

File: app/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -20,6 +20,7 @@
 		42127D9B2C283C310005AB40 /* OneSignalExtension.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = 42127D9A2C283C310005AB40 /* OneSignalExtension.xcframework */; };
 		42127D9E2C283C400005AB40 /* OneSignalOSCore.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = 42127D9D2C283C400005AB40 /* OneSignalOSCore.xcframework */; };
 		42127DA12C283C4E0005AB40 /* OneSignalOutcomes.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = 42127DA02C283C4E0005AB40 /* OneSignalOutcomes.xcframework */; };
+		42476A842C5E36DA00426330 /* libsqlite3.tbd in Frameworks */ = {isa = PBXBuildFile; fileRef = 42476A832C5E369200426330 /* libsqlite3.tbd */; };
```
Comment (josancamon19) [inline app/ios/Runner.xcodeproj/project.pbxproj]: What are this changes for?

**Review by josancamon19 — CHANGES_REQUESTED** [review]

File: app/lib/pages/home/firmware_update.dart (hunk 2 above)
Comment (mdmohsin7) [reply app/lib/pages/home/firmware_update.dart:1]: Yes I built an APK and tested as well

File: app/lib/pages/home/firmware_mixin.dart (hunk 3 above)
Comment (mdmohsin7) [reply app/lib/pages/home/firmware_mixin.dart:1]: The device becomes useless. I tried closing the app during the installation process, and now the app doesn't recognise the device at all. It won't show up at all. I had to manually upload the firmware for it to work again.

File: app/ios/Runner.xcodeproj/project.pbxproj (hunk 4 above)
Comment (mdmohsin7) [reply app/ios/Runner.xcodeproj/project.pbxproj]: This was related to the flutter_downloader package which I removed

File: app/lib/pages/home/firmware_mixin.dart (hunk 3 above)
Comment (francip) [reply app/lib/pages/home/firmware_mixin.dart:1]: Might be nice to have a link in the app to a webpage with instructions how to recover in such case.

Comment (mdmohsin7) [reply app/lib/pages/home/firmware_mixin.dart:1]: We can definitely do that, but where should we have that link? Under guides section maybe? We can have multiple links in here.

Screenshot 2024-08-10 at 12 56 41 AM

**Review by josancamon19 — APPROVED** [review]

**Comment by josancamon19** [comment]
Merging for now will fix on main

**Pull request merged**